### PR TITLE
Integrate SwiftLint into iOS Sync Engine Project

### PR DESCRIPTION
**Pull request by agisilaos**
## What's new in this PR?

Integrating SwiftLint into the iOS Sync Engine project. The `.swiftlint.yml` file is based on the file that's been added on the `wire-ios` repository. I've also added a GitHub action for SwiftLint 

## Notes

https://wearezeta.atlassian.net/jira/software/projects/IC/boards/100?selectedIssue=IC-6

### CODE REVIEW

File: .github/workflows/swiftlint.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: SwiftLint
+
+on:
+  pull_request:
+    paths:
+      - '.github/workflows/swiftlint.yml'
+      - '.swiftlint.yml'
```
Comment (billypchan) [inline .github/workflows/swiftlint.yml:7]: can we just keep one copy of rules?

**Comment by billypchan** [comment]
the branch name should be standardise like `fix/integrate-swiftlint-into-the-sync-engine-project`, and it would be better to create a jira ticket for a record.

**Comment by billypchan** [comment]
I would suggest create a demo PR to check how does the Github action works.

**Comment by agisilaos** [comment]
> the branch name should be standardise like fix/integrate-swiftlint-into-the-sync-engine-project, and it would be better to create a jira ticket for a record.

There's already a Jira ticket for the issue.

File: .github/workflows/swiftlint.yml (hunk 1 above)
Comment (agisilaos) [reply .github/workflows/swiftlint.yml:7]: This is how it's been set up in the `wire-ios` repo.

**Comment by agisilaos** [comment]
Closing this one to make the appropriate changes @billypchan made regarding the name of the branch etc and will open a new one.

**Pull request closed without merging**